fdk.go

```diff
@@ -107,6 +111,84 @@ func doHTTP(handler Handler, ctx context.Context, in io.Reader, out io.Writer) {
 	}
 }
 
+func doJSON(handler Handler, ctx context.Context, in io.Reader, out io.Writer) {
+	var buf bytes.Buffer
+	hdr := make(http.Header)
+
+	for {
+		doJSONOnce(handler, ctx, in, out, &buf, hdr)
+	}
+}
+
+type jsonio struct {
+	Body        string `json:"body"`
+	ContentType string `json:"content_type"`
+}
+
+type callRequestHTTP struct {
+	Type       string      `json:"type"`
+	RequestURL string      `json:"request_url"`
+	Headers    http.Header `json:"headers"`
+}
+
+type jsonIn struct {
+	jsonio
+	CallID   string           `json:"call_id"`
+	Protocol *callRequestHTTP `json:"protocol"`
+}
+
+type callResponseHTTP struct {
+	StatusCode int         `json:"status_code,omitempty"`
+	Headers    http.Header `json:"headers,omitempty"`
+}
+
+type jsonOut struct {
+	jsonio
+	Protocol *callResponseHTTP `json:"protocol,omitempty"`
+
+	io.Writer
+}
+
+func (out *jsonOut) Write(p []byte) (n int, err error) {
+	out.Body += string(p)
+	return len(p), nil
+}
+
+func (out *jsonOut) WriteStatus(status int) {
+	out.Protocol.StatusCode = status
+}
+
+func doJSONOnce(handler Handler, ctx context.Context, in io.Reader, out io.Writer, buf *bytes.Buffer, hdr http.Header) {
+	buf.Reset()
+	resetHeaders(hdr)
+
+	resp := &jsonOut{
+		jsonio: jsonio{
+			ContentType: "application/json",
+		},
+		Protocol: &callResponseHTTP{
+			StatusCode: 200,
+		},
+		Writer: buf,
+	}
+
+	req := &jsonIn{}
```

Comment:
nit: can save an alloc here by using a var and referencing it (1 alloc each loop)

Reply:
makes sense